common/utils/cpp_circular_buffer.h

```diff
@@ -156,6 +156,60 @@ class CircularBuffer {
    * @return bool: True if buffer is full, false otherwise.
    */
   bool full() const { return this->count() == Capacity; }
+
+  /**
+   * Get the average of all elements in the buffer.
+   * 
+   * @return Type: The average of all elements in the buffer.
+   */
+  Type average() {
+    Type sum = this->sum();
+    return sum / this->count();
+  }
+
+  /**
+   * Get the median of all elements in the buffer.
+   * 
+   * @return Type: The median of all elements in the buffer.
+   */
+  Type median() {
+    size_t size = this->count();
+    std::vector<Type> linearData(size);
```

Comment:
Ouch oof my dynamic memory allocation (please avoid if possible)

Reply:
What Matt means is that an `std::vector` uses heap-allocated dynamic memory. We want to avoid this pretty much completely in our embedded software because it can result in memory leaks if you aren't careful, and even if you are it's slower and less deterministically analyzable than stack allocation.

Can you change this to use a statically allocated array of size `Capacity`? We know that will always be enough to hold all of the elements.

Reply:
Option B, sort an array of pointers to members?

Reply:
That doesn't get us around needing to allocate the array somewhere, right? Definitely sorting pointers should be preferred vs copying objects themselves though.

Reply:
Yeah, gotcha. Statically allocated a fixed size array of Type*[]s somewhere would be required, as a member variable, I agree.

Reply:
And yeah I agree with something Sam said earlier too. This extra math stuff might best live in a new class that extends this base one? IDK

Reply:
changed to a array of size capacity, also moved everything to a different class, not sure about the naming though.